pallets/connectors/src/message.rs

```diff
@@ -76,244 +89,293 @@ impl<Domain: Encode, PoolId: Encode, TrancheId: Encode, Balance: Encode, Rate: E
-				message.append(&mut address.encode());
-				message.append(&mut valid_until.to_be_bytes().to_vec());
-
-				message
-			}
+			} => encoded_message(
+				self.call_type(),
+				vec![
+					encode_be(pool_id),
+					tranche_id.encode(),
+					address.encode(),
+					valid_until.to_be_bytes().to_vec(),
+				],
+			),
 			Message::Transfer {
 				pool_id,
 				tranche_id,
 				domain,
 				address,
 				amount,
-			} => {
-				let mut message: Vec<u8> = vec![];
-				message.push(self.call_type());
-				message.append(&mut to_be(pool_id));
-				message.append(&mut tranche_id.encode());
-				message.append(&mut domain.encode());
-				message.append(&mut address.encode());
-				message.append(&mut to_be(amount));
-
-				message
-			}
+			} => encoded_message(
+				self.call_type(),
+				vec![
+					encode_be(pool_id),
+					tranche_id.encode(),
+					domain.encode(),
+					address.encode(),
+					encode_be(amount),
+				],
+			),
 		}
 	}
 }
 
-/// Encode a value in its big-endian representation. We use this for number types to make
-/// sure they are encoded the way they are expected to be decoded on the Solidity side.
-fn to_be(x: impl Encode) -> Vec<u8> {
-	let mut output = x.encode();
-	output.reverse();
-	output
+fn encoded_message(call_type: u8, fields: Vec<Vec<u8>>) -> Vec<u8> {
+	let mut message: Vec<u8> = vec![];
+	message.push(call_type);
+	for x in fields {
+		message.append(&mut x.clone());
+	}
+
+	message
 }
 
 #[cfg(test)]
 mod tests {
+	use cfg_primitives::{Balance, PoolId, TrancheId};
 	use cfg_types::fixed_point::Rate;
-	use codec::Encode;
+	use cfg_utils::vec_to_fixed_array;
+	use codec::{Decode, Encode};
 	use hex::FromHex;
 	use sp_runtime::traits::One;
 
-	use crate::Message;
-
-	type PoolId = u64;
-	type TrancheId = [u8; 16];
-	type Balance = cfg_primitives::Balance;
-
-	pub mod encode {
-		use cfg_utils::vec_to_fixed_array;
-
-		use super::*;
-		use crate::{Domain, DomainAddress};
-
-		#[test]
-		fn invalid() {
-			let msg = Message::<Domain, PoolId, TrancheId, Balance, Rate>::Invalid;
-			assert_eq!(msg.encode(), vec![msg.call_type()]);
-			assert_eq!(msg.encode(), vec![0]);
-		}
-
-		#[test]
-		fn encoding_domain() {
-			use crate::Encode;
-
-			// The Centrifuge substrate chain
-			assert_eq!(
-				hex::encode(Domain::Centrifuge.encode()),
-				"000000000000000000"
-			);
-			// Ethereum MainNet
-			assert_eq!(hex::encode(Domain::EVM(1).encode()), "010000000000000001");
-			// Moonbeam EVM chain
-			assert_eq!(
-				hex::encode(Domain::EVM(1284).encode()),
-				"010000000000000504"
-			);
-			// Avalanche Chain
-			assert_eq!(
-				hex::encode(Domain::EVM(43114).encode()),
-				"01000000000000a86a"
-			);
-		}
+	use super::*;
+	use crate::{Domain, DomainAddress};
 
-		#[test]
-		fn add_pool_zero() {
-			let msg = Message::<Domain, PoolId, TrancheId, Balance, Rate>::AddPool { pool_id: 0 };
-			let encoded = msg.encode();
+	pub type ConnectorMessage = Message<Domain, PoolId, TrancheId, Balance, Rate>;
```

Comment:
I'm worried that because `Message` is something generic for these types, we could be testing that it's working with the test types, but different configurations of the connector pallet could lead to different serializations. i.e. How a `Rate` is encoding is defined by the `Rate` used in the configuration.

Reply:
Yeah that's a good point. What safe-guards us is the `expected_hex` values, that's why it's important to have those in the tests 👍